pkg/project/auth/cache.go

```diff
@@ -172,6 +178,30 @@ func (ac *AuthorizationCache) Run(period time.Duration) {
 	go utilwait.Forever(func() { ac.synchronize() }, period)
 }
 
+func (ac *AuthorizationCache) AddWatcher(watcher CacheWatcher) {
+	ac.watcherLock.Lock()
+	defer ac.watcherLock.Unlock()
+
+	ac.watchers = append(ac.watchers, watcher)
+}
+
+func (ac *AuthorizationCache) RemoveWatcher(watcher CacheWatcher) {
+	ac.watcherLock.Lock()
+	defer ac.watcherLock.Unlock()
+
+	lastIndex := len(ac.watchers) - 1
+	for i := 0; i < len(ac.watchers); i++ {
+		if ac.watchers[i] == watcher {
+			if i < lastIndex {
+				// if we're not the last element, shift
+				copy(ac.watchers[i:], ac.watchers[i+1:])
```

Comment:
@liggitt I'm not completely comfortable with these operations.  This the best way?